### ISSUE TITLE
[FORMATTING] Querying a semi-structured data with an element name that is a reserved clause is incorrect

### ISSUE THREAD

**Issue by chaupow** [sql-formatter-org/sql-formatter#706]
Hi! 👋 

**Input data**

Which SQL and options did you provide as input?
with `"language": "snowflake"`

```sql
SELECT
  details:type
FROM
  transactions
```

**Expected Output**

```sql
SELECT
  details:type
FROM
  transactions
```

**Actual Output**

```sql
SELECT
  details:
type
FROM
  transactions
```

**Usage**

We're using snowflake and unfortunately some of our JSON fields have the key `type` which is identified as a reserved clause and incorrectly put in the next line.

We can prevent this by using `details:"type"` but it would be great if it would work without quotes as well.

This is especially an issue because we would love to use `"keywordCase": "upper"` but this breaks the query because key names are case sensitive and `details:TYPE` fails to query the intended lookup.

- How are you calling / using the library?
```
sqlFormatter.format(sqlContent, {"language": "snowflake"});
```

- What SQL language(s) does this apply to?
`snowflake`

- Which SQL Formatter version are you using?
`15.1.2`

Thank youuuuuuuuu 🙏

**Comment by nene** [comment]
Thanks for reporting.

Although [I apparently have documented](https://github.com/sql-formatter-org/sql-formatter/wiki/arrays-and-maps) this `:`-syntax of Redshift, I haven't actually implemented any support for it.

Interestingly I just recently fixed [a similar issue for Transact-SQL](https://github.com/sql-formatter-org/sql-formatter/issues/700). So most likely some similar solution could be done in here.

**Comment by nene** [comment]
I was mistaken. This syntax was supported, but as the issue title says, the problem was only the case when a keyword occurred after `:`.

Now treating `:` similarly to `.` that's commonly used as a `table.column` separator.

**Comment by nene** [comment]
Fixed in 15.1.3

**Comment by chaupow** [comment]
Oh wow this was amazing quick!! ⚡✨ 

Thanks so much for fixing so promptly and pushing a new version out 🥹
I really appreciate it!
